### PR TITLE
Support running e2e tests on MacOS

### PR DESCRIPTION
**Pull request by ethan256**

### CODE REVIEW

**Comment by wu-sheng** [comment]
Should we add a new CI env of MacOS to verify this?

**Comment by mrproliu** [comment]
> Should we add a new CI env of MacOS to verify this?

Yes. @ethan256 Please add a simple GHA to verify the plugin test on MacOS. a single plugin testing is enough, such as the native HTTP plugin.

File: test/plugins/run.sh [hunk 1]
```diff
@@ -155,16 +155,24 @@ for framework_version in $frameworks; do
...
-  sed -i "s/^go [0-9]*\.[0-9]*/go ${go_version}/" go.mod
   # append the module name(for go1.20, module name cannot be same in same workspace)
   mod_case_name=$(echo "${case_name}" | sed 's/\//_/g; s/\./_/g; s/-/_/g')
   mod_name=$(head -n 1 go.mod | cut -d " " -f 2)
-  sed -i "s/^module /module ${mod_case_name}\//" go.mod
-  find . -name "*.go" -type f -exec sed -i "s|${mod_name}|${mod_case_name}/${mod_name}|g" {} \;
-
-  # ajust the plugin replace path
-  sed -i -E '/^replace/ s#(\.\./)#\1../#' go.mod
+  if [ $(uname) = 'Darwin' ]; then
```
Comment (lujiajing1126) [inline test/plugins/run.sh:161]: I would suggest moving conditional check to the top of the file, for example with 'alias' or creating a new function?

**Review by lujiajing1126 — CHANGES_REQUESTED** [review]

File: test/plugins/run.sh (hunk 1 above)
Comment (mrproliu) [reply test/plugins/run.sh:161]: I think a new function is more reasonable. @ethan256 what do you think?

File: .github/workflows/plugin-tests.yaml [hunk 2]
```diff
@@ -85,18 +98,58 @@ jobs:
...
       - test
+      - test-on-macos
     runs-on: ubuntu-latest
     timeout-minutes: 10
     steps:
       - name: Merge Requirement
         run: |
           if [[ ${{ needs.test.result }} != 'success' ]]; then
             exit -1
-          fi
```
Comment (mrproliu) [inline .github/workflows/plugin-tests.yaml:102]: Please add the MacOs test result checker.

**Review by lujiajing1126 — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
@lujiajing1126 Please make sure you have commit logs reset when squash. It is very crazy to see this on commit log.

image

Most are only noises.

**Comment by lujiajing1126** [comment]
> @lujiajing1126 Please make sure you have commit logs reset when squash. It is very crazy to see this on commit log.
> 
> image
> Most are only noises.

I will notice next time

**Pull request merged**